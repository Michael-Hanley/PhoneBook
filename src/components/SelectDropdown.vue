<template>
  <div>
    <div>
      <label for="select"><slot></slot></label>
    </div>
    <select
      id="select"
      class="select"
      @change="this.onChange"
      v-model="localValue"
    >
      <option v-for="(option, index) in options" :key="index">
        {{ option }}
      </option>
    </select>
  </div>
</template>

<script>
export default {
  name: "SelectDropdown",
  props: {
    options: Array,
    value: String,
  },
  data() {
    return {
      localValue: this.value,
    };
  },
  watch: {
    localValue: function () {
      this.localValue = this.value;
    },
  },
  methods: {
    onChange: function () {
      console.log(this.localValue);
      this.$emit("input", this.localValue);
    },
  },
};
</script>

<style scoped>
.select {
  box-sizing: border-box;
  background-color: transparent;
  width: 100%;
  padding: 5px;
  margin: 5px;
  border: none;
  font-size: inherit;
  text-align: left;
  outline: none;
  border-bottom: 1px solid gray;
  cursor: pointer;
}
.select:focus {
  margin-bottom: 4px;
  border-bottom: 2px solid black;
}
</style>
