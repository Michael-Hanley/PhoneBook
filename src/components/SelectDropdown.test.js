import { shallowMount } from '@vue/test-utils';
import SelectDropdown from './SelectDropdown.vue';

let wrapper;
let testOptions = ['test-option-1', 'test-option-2']
describe('SelectDropdown', () => {
  beforeEach(() => {
    wrapper = shallowMount(SelectDropdown, {
      slots: {
        default: 'test'
      },
      propsData: {
        options: [...testOptions],
        value: testOptions[0]
      }
    })
  });
  it('should display the proper option when selected', () => {
    expect(wrapper.vm.localValue).toBe(testOptions[0]);

    wrapper.findAll('select > option').at(1).element.selected = true;
    wrapper.find('select').trigger('change');

    expect(wrapper.vm.localValue).toBe(testOptions[1])
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});