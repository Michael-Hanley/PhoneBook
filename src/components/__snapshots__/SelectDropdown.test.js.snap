// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown should render correctly 1`] = `
<div>
  <div><label for="select">test</label></div> <select id="select" class="select">
    <option>
      test-option-1
    </option>
    <option>
      test-option-2
    </option>
  </select>
</div>
`;
